doc/CounterChecks.markdown

```diff
@@ -0,0 +1,224 @@
+# PrivCount Counter Checks
+
+PrivCount checks counter configurations when it loads counters, and before
+a round starts.
+
+## Initial Checks
+
+PrivCount performs various counter sanity checks to guard against errors.
+
+PrivCount checks the following counter properties:
+* every time the config is loaded (Tally Server), or
+* at the start of every round (Share Keeper, Data Collector).
+
+If these checks fail, the configuration is rejected and no collection occurs.
+
+### Counters
+
+Each counter counts events in a set of bins.
+
+* Each counter name must be in the list of valid counter names  
+  (This list can have counters added to it at runtime)
+* Counters for bins and sigmas must match
+
+### Noise
+
+Each counter has noise added to it when it is initialised by the data
+collector.
+
+#### Noise Weights
+
+The noise weight reflects how much a client's activity is expected to change
+the counter.
+
+* Each noise weight is a non-negative decimal value
+* The total noise weight is less than the maximum counter value
+
+#### Sigmas
+
+The sigma determines how much noise is added to a counter.
+
+* Sigma counters must have a sigma element
+* Each sigma is a non-negative decimal value
+
+### Bins
+
+The bins determine how events are aggregated in the counter.
+
+* Bin counters must have a bins element
+* Bins have lower and upper bounds which are decimal values  
+  (including positive and negative infinity)
+* The upper bound is strictly greater than the lower bound
+* The bins in a counter must not overlap (but there may be gaps)
+
+### Periods
+
+The periods determine how often nodes communicate, and how long rounds last.
+
+#### Collect Period
+
+Data Collectors collect events for collect period seconds.
```

Comment:
I would add that the events are collected from Tor and that PrivCount increments the counters corresponding to the events it receives.